packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,285 @@
+            return;
+          }
+        }
+        nextFocus.focus();
+      }
+    };
+  }
+
+  handleSelection(event, option) {
+    if (this.props.onChange && option.props.value !== this.state.value) {
+      this.props.onChange(event, option.props.value);
+    }
+
+    let newValue = null;
+    if (!this.props.value) {
+      newValue = option.props.value;
+    }
+
+    this.setState({ selectedValue: newValue, isOpen: false });
+  }
+
+  handleOnClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleOnKeyDown(event) {
+    if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.SPACE || event.nativeEvent.keyCode === SelectUtils.KEYCODES.ENTER) {
+      this.setState({ isOpen: true });
+    }
+  }
+
+  handleOnRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  render() {
+    const {
+      onChange,
+      name,
+      required,
+      defaultValue,
+      value,
+      isInvalid,
+      disabled,
+      children,
+      ...customProps
+    } = this.props;
+
+    const { intl } = this.context;
+
+    const attributes = Object.assign({}, customProps);
+    const selectClasses = cx([
+      'select',
+      { invalid: isInvalid },
+      { 'is-disabled': disabled },
+      { open: this.state.isOpen },
+      attributes.className,
+    ]);
+
+    if (required) {
+      attributes['aria-required'] = 'true';
+    }
+
+    attributes.tabIndex = disabled ? '-1' : '0';
+    attributes['aria-disabled'] = disabled;
+
+    let selectedValue;
+    let display;
+    let defaultOption;
+    let isDefaultDisplay = false;
+    const clonedChildren = React.Children.map(children, (child) => {
+      let isSelected = false;
+
+      if (child.props.value === value || child.props.value === this.state.selectedValue) {
+        selectedValue = child.props.value;
+        display = child.props.display || child.props.children;
+        isSelected = true;
+      }
+
+      return React.cloneElement(child, {
+        onClick: this.wrapOnClick(child),
+        onKeyDown: this.wrapOnKeyDown(child),
+        isSelected,
+      });
+    });
+
+    if (!selectedValue) {
+      display = intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
+      selectedValue = '';
+      defaultOption = <SelectOption value={selectedValue} display={display} isSelected className={cx('default-option')} />;
+      isDefaultDisplay = true;
+    }
+
+    return (
+      <div>
+        <Popup
+          contentAttachment="top left"
+          targetAttachment="bottom left"
+          contentHeight="auto"
+          contentWidth="auto"
+          onRequestClose={this.handleOnRequestClose}
+          attachmentBehavior="flip"
+          isOpen={this.state.isOpen}
+          targetRef={this.getTargetRef}
+          classNameContent={cx('select-menu')}
+        >
+          <SelectMenu>
+            {defaultOption}
+            {clonedChildren}
+          </SelectMenu>
+        </Popup>
+        {/* eslint-disable jsx-a11y/no-static-element-interactions */}
```

Comment:
Do we have a better alternative for attaching on click handlers to some interactive html element, instead of non interactive containers to satisfy eslint? I'm running into this same issue with the control components being wrapped in container elements.

Reply:
I workaround to possibly quiting the eslint could be creating an object that contained the onClick and onkeyDown like 

```cont callbacks = { onClick: () => {}, onKeyDown: ()=>{} }``` 
and then spreading them on the div like 
```<div {...callbacks}``` 
I remember @dkasper-was-taken mentioning this before.

Reply:
Disabling this eslint rule here is only needed because there is a bug in the version of eslint that we're using. Adding an appropriate role to the element is supposed to satisfy the rule, but it doesn't pick it up. https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/272

I really don't think that we should be using work arounds to trick eslint. If we don't think the rule is appropriate we should really just turn off the rule.